test/s3path.jl

```diff
@@ -355,6 +355,30 @@ end
     rm(json_path)
 end
 
+@testset "`tryparse`" begin
+    get_values = (str) -> begin
+        p = S3Path(str)
+        return (p.segments, p.root, p.drive, p.isdirectory, p.version)
+    end
+
+    @test isequal(get_values("s3://my_bucket/prefix/that/is/fun"),
+                  (("prefix", "that", "is", "fun"), "/", "s3://my_bucket", false, nothing))
+    @test isequal(get_values("s3://my_bucket/prefix/that/is/fun/"),
+                  (("prefix", "that", "is", "fun"), "/", "s3://my_bucket", true, nothing))
+    @test isequal(get_values("s3://my_bucket/"), ((), "/", "s3://my_bucket", true, nothing))
+    @test isequal(get_values("s3://my_bucket"), ((), "", "s3://my_bucket", true, nothing))
+    @test isequal(get_values("s3://my_bucket/prefix/that/is/fun?versionId=xyz"),
+                  (("prefix", "that", "is", "fun"), "/", "s3://my_bucket", false, "xyz"))
+    @test isequal(get_values("s3://my_bucket/prefix/that/is/fun/?versionId=xyz"),
+                  (("prefix", "that", "is", "fun"), "/", "s3://my_bucket", true, "xyz"))
+    @test isequal(get_values("s3://my_bucket/?versionId=xyz"), ((), "/", "s3://my_bucket", true, "xyz"))
+    @test isequal(get_values("s3://my_bucket?versionId=xyz"),((), "", "s3://my_bucket", true, "xyz"))
+    @test isequal(get_values("s3://my_bucket/prefix/that/is/fun/?versionId=xyz&radtimes=foo"),
+                  (("prefix", "that", "is", "fun"), "/", "s3://my_bucket", true, "xyz"))
+    @test isequal(get_values("s3://my_bucket/prefix/that/is/fun/?radtimes=foo&versionId=xyz"),
+                (("prefix", "that", "is", "fun"), "/", "s3://my_bucket", true, "xyz"))
+end
```

Comment:
These tests are rather hard to understand at first glance. I think we can do better if we use the inner `S3Path` constructor and make use of `==`. We should also make use of vertical whitespace to avoid this being a wall of text:
```suggestion
@testset "tryparse" begin
    cfg = global_aws_config()

    @test S3Path("s3://my_bucket/prefix/that/is/fun") ==
          S3Path(("prefix", "that", "is", "fun"), "/", "s3://my_bucket", false, cfg, nothing)

    @test S3Path("s3://my_bucket/prefix/that/is/fun/") ==
          S3Path(("prefix", "that", "is", "fun"), "/", "s3://my_bucket", true, cfg, nothing)

    @test S3Path("s3://my_bucket/") ==
          S3Path((), "/", "s3://my_bucket", true, cfg, nothing)

    @test S3Path("s3://my_bucket") ==
          S3Path((), "", "s3://my_bucket", true, cfg, nothing)

    @test S3Path("s3://my_bucket/prefix/that/is/fun?versionId=xyz") ==
          S3Path(("prefix", "that", "is", "fun"), "/", "s3://my_bucket", false, cfg, "xyz")

    @test S3Path("s3://my_bucket/prefix/that/is/fun/?versionId=xyz") ==
          S3Path(("prefix", "that", "is", "fun"), "/", "s3://my_bucket", true, cfg, "xyz")

    @test S3Path("s3://my_bucket/?versionId=xyz") ==
          S3Path((), "/", "s3://my_bucket", true, cfg, "xyz")

    @test S3Path("s3://my_bucket?versionId=xyz") ==
          S3Path((), "", "s3://my_bucket", true, cfg, "xyz")

    @test S3Path("s3://my_bucket/prefix/that/is/fun/?versionId=xyz&radtimes=foo") ==
          S3Path(("prefix", "that", "is", "fun"), "/", "s3://my_bucket", true, cfg, "xyz")

    @test S3Path("s3://my_bucket/prefix/that/is/fun/?radtimes=foo&versionId=xyz") ==
          S3Path(("prefix", "that", "is", "fun"), "/", "s3://my_bucket", true, cfg, "xyz")
end
```
Aside: Based upon these changes it probably makes sense to have the `version` field precede `config`. Best to do that before we actually tag a release